modules/tinymce/src/themes/silver/main/ts/ui/context/ContextToolbar.ts

```diff
@@ -0,0 +1,276 @@
+      lastPos: lastPosition.get,
+      lastElement: lastElement.get,
+      isReposition: () => Optionals.is(lastTrigger.get(), TriggerCause.Reposition),
+      bounds: lastBounds.get
+    });
+    return Merger.deepMerge(anchorage, anchorLayout);
+  };
+
+  const launchContext = (toolbarApi: Array<ContextType>, elem: Optional<Element>) => {
+    launchContextToolbar.stop();
+
+    // If a mobile context menu is open, don't launch else they'll probably overlap. For android, specifically.
+    if (isTouch() && backstage.isContextMenuOpen()) {
+      return;
+    }
+
+    const toolbarSpec = buildToolbar(toolbarApi);
+    const sElem = elem.map(SugarElement.fromDom);
+
+    // TINY-4495 ASSUMPTION: Can only do toolbarApi[0].position because ContextToolbarLookup.filterToolbarsByPosition
+    // ensures all toolbars returned by ContextToolbarLookup have the same position.
+    // And everything else that gets toolbars from elsewhere only returns maximum 1 toolbar
+    const anchor = getAnchor(toolbarApi[0].position, sElem);
+    lastTrigger.set(TriggerCause.NewAnchor);
+
+    const contextBarEle = contextbar.element;
+    Css.remove(contextBarEle, 'display');
+    InlineView.showWithinBounds(contextbar, anchor, wrapInPopDialog(toolbarSpec), () => Optional.some(getBounds()));
+
+    // IMPORTANT: This must be stored after the initial render, otherwise the lookup of the last element in the
+    // anchor placement will be incorrect as it'll reuse the new element as the anchor point.
+    sElem.fold(lastElement.clear, lastElement.set);
+    lastPosition.set(Boxes.box(contextbar.element));
+
+    // It's possible we may have launched offscreen, if so then hide
+    if (shouldContextToolbarHide()) {
+      Css.set(contextBarEle, 'display', 'none');
+    }
+  };
+
+  const launchContextToolbar = Delay.debounce(() => {
+    // Don't launch if the editor doesn't have focus or has been destroyed
+    if (!editor.hasFocus() || editor.removed) {
+      return;
+    }
+
+    const scopes = getScopes();
+    ToolbarLookup.lookup(scopes, editor).fold(
+      close,
+
+      (info) => {
+        launchContext(info.toolbars, Optional.some(info.elem.dom));
+      }
+    );
+  }, 0);
+
+  editor.on('init', () => {
+    editor.on('remove', close);
+    editor.on('ScrollContent ScrollWindow ObjectResized ResizeEditor longpress', hideOrRepositionIfNecessary);
+
+    // FIX: Make it go away when the action makes it go away. E.g. deleting a column deletes the table.
+    editor.on('click keyup focus SetContent', launchContextToolbar);
+
+    editor.on(hideContextToolbarEvent, close);
+    editor.on(showContextToolbarEvent, (e) => {
+      const scopes = getScopes();
+      // TODO: Have this stored in a better structure
+      Obj.get(scopes.lookupTable, e.toolbarKey).each((ctx) => {
+        // ASSUMPTION: this is only used to open one specific toolbar at a time, hence [ctx]
+        launchContext([ ctx ], Optionals.someIf(e.target !== editor, e.target));
```

Comment:
As mentioned on slack, this was a bug as previously it did `e as Element` however `e` was an editor event. Luckily we have no code path that triggers this event where the target isn't the editor. I didn't include a changelog entry for this since it's not something that can be triggered or impacted anyone currently.